### PR TITLE
Add e2e test waitFor functions for common policy conditions

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
The main conditions e2e tests wait for are policy Available/Degraded,
this commit wrap them at a waitForAvailable and waitForDegraded functions.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by qinqon** [comment]
Found issue at brext test, we may need to retry more:

```
2020-01-03T10:23:06.588Z	INFO	enactmentconditions	NotifySuccess	{"enactment": "node02.default-network"}
2020-01-03T10:23:06.592Z	ERROR	enactmentconditions	Error notifying state Success	{"enactment": "node02.default-network", "error": "Put https://10.96.0.1:443/apis/nmstate.io/v1alpha1/nodenetworkconfigurationenactments/node02.default-network/status: read tcp 192.168.66.128:50318->10.96.0.1:443: read: connection reset by peer"}
github.com/go-logr/zapr.(*zapLogger).Error
	kubernetes-nmstate/vendor/github.com/go-logr/zapr/zapr.go:128
github.com/nmstate/kubernetes-nmstate/pkg/controller/nodenetworkconfigurationpolicy/enactmentconditions.(*EnactmentConditions).NotifySuccess
	kubernetes-nmstate/pkg/controller/nodenetworkconfigurationpolicy/enactmentconditions/conditions.go:85
github.com/nmstate/kubernetes-nmstate/pkg/controller/nodenetworkconfigurationpolicy.(*ReconcileNodeNetworkConfigurationPolicy).Reconcile
	kubernetes-nmstate/pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go:203
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	kubernetes-nmstate/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	kubernetes-nmstate/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:192
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	kubernetes-nmstate/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:171
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	kubernetes-nmstate/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	kubernetes-nmstate/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	kubernetes-nmstate/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
2020-01-03T10:23:06.592Z	INFO	policyconditions	enactments count: {failed: 0, progressing: 2, available: 0}	{"policy": "default-network"}
```

File: pkg/helper/client.go [hunk 1]
```diff
@@ -196,6 +199,30 @@ func ping(target string, timeout time.Duration) (string, error) {
 	})
 }
 
+func checkApiServerConnectivity(timeout time.Duration) error {
+
```
Comment (phoracek) [inline pkg/helper/client.go]: Please drop the empty line

File: pkg/helper/client.go [hunk 2]
```diff
@@ -196,6 +199,30 @@ func ping(target string, timeout time.Duration) (string, error) {
...
+		if err != nil {
+			return false, errors.Wrap(err, "getting config")
+		}
+		// Since we are going to retrieve Nodes default schema is good
+		// enough
+		client, err := client.New(config, client.Options{})
+		if err != nil {
+			return false, errors.Wrap(err, "creating new custom client")
+		}
+		nodes := corev1.NodeList{}
```
Comment (phoracek) [inline pkg/helper/client.go]: Nodes are huge, they contain list of all their pods, resources etc. We should get our own Enactment we created (or some meta object on kubernetes API, e.g. cluster version, if such a thing exists).

File: pkg/helper/client.go [hunk 3]
```diff
@@ -196,6 +199,30 @@ func ping(target string, timeout time.Duration) (string, error) {
...
+		err = client.List(context.TODO(), &nodes)
+		if err != nil {
+			return false, errors.Wrap(err, "getting nodes failed")
```
Comment (phoracek) [inline pkg/helper/client.go]: This would be really confusing for users. "reaching to apiserver failed" maybe?

File: pkg/helper/client.go [hunk 4]
```diff
@@ -260,6 +287,11 @@ func ApplyDesiredState(desiredState nmstatev1alpha1.State) (string, error) {
 		return pingOutput, rollback(fmt.Errorf("error pinging external address after network reconfiguration -> error: %v, currentState: %s", err, currentState))
 	}
 
+	err = checkApiServerConnectivity(defaultGwProbeTimeout * time.Second)
```
Comment (phoracek) [inline pkg/helper/client.go]: We need to change the name of `defaultGwProbeTimeout` so it is shared for both and substract from it after we ping the GW. Or we can introduce another constant just for the apiserver.

Comment (qinqon) [reply pkg/helper/client.go]: Adding new constant.

File: pkg/helper/client.go (hunk 2 above)
Comment (qinqon) [reply pkg/helper/client.go]: We can try the counterpart of ```kubectl get namespace default```

**Pull request merged**